operator/pkg/helmreconciler/rendering.go

```diff
@@ -209,9 +199,7 @@ func (h *HelmReconciler) ProcessManifest(manifest manifest.Manifest) (int, error
 		}
 		log.Infof("Adding object %s to cache.", obj.Hash())
 		// Update the cache with the latest object.
-		objectCacheMu.Lock()
-		objectCache[obj.Hash()] = obj
-		objectCacheMu.Unlock()
+		h.objectCache[obj.Hash()] = obj
```

Comment:
Because ProcessManifest may be called from multiple threads, doesn't this still need to be protected by a mutex?

Reply:
i don't think so because each thread will have a copy of h. which also makes me realize that i have to go back to the global cache since unfortunately the helm reconciler is created each time we receive the CR update. we could keep it hanging around but that will be even more messy than having a global cache with locking.

Reply:
It probably should be initialized with the original cache, to help in processing of the updates, but should be replaced with new versions when the update is complete, but before going through the prune.

If the global cache accounts for resources in different namespaces, belonging to different ICPs, then global is probably fine.  The key, as mentioned above is to ensure the cache correctly reflects what's actually been installed (i.e. no stale entries).

Reply:
yes, i think there could be a problem with cache not being pruned - objects will be pruned the first time because pruning is based on the manifest list not the cache but the next time around the same object may not be written to the API server if it was pruned the last time because it's still in the cache. 
you're correct that we shouldn't have namespace overlap between CRs, but even if we do it just results in some unnecessary writes to the server, not data consistency problems. so i think we are good here.